lib/mongoid/errors/invalid_estimated_count_criteria.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+# encoding: utf-8
+
+module Mongoid
+  module Errors
+
+    # Creates the exception raised when trying to call estimated_count
+    # on a filtered criteria.
+    #
+    # @param [ String ] class_name The class name.
+    #
+    # @api private
```

Comment:
You can have api private on the constructor but the class overall is not private because it is used for exceptions reported to applications.

Reply:
So maybe move this entire docstring to the constructor, since it seems to apply to the constructor and not the class.